src/Lights/Shadows/shadowGenerator.ts

```diff
@@ -885,10 +885,23 @@ export class ShadowGenerator implements IShadowGenerator {
             if (this._filter === ShadowGenerator.FILTER_PCF) {
                 engine.setColorWrite(false);
             }
+            if (this._scene.getSceneUniformBuffer().useUbo) {
+                const sceneUBO = this._scene.getSceneUniformBuffer();

```

Comment:
Why do we need this change?

Reply:
When using a `StandardMaterial` as a base material for the custom depth material as in my sample PG, the view/projection transformations I got in the shader were the ones from the scene, not the ones I bound with:
```typescript
        effect.setMatrix("viewProjection", this.getTransformMatrix());
        effect.setMatrix("view", this._viewMatrix);
        effect.setMatrix("projection", this._projectionMatrix);
```
I had to put the code above to get the right matrices. I don't have this problem when using the ShaderMaterial class. I think standard / PBR materials use UBO by default when available.

Reply:
oki!